### PR TITLE
Remove default `SvgTheme` argument preventing `DefaultSvgTheme` works correctly

### LINKED ISSUE #484: [BUG]: flutter_svg `DefaultSvgTheme` not working
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

5.4.0

### Command type

build_runner (Default)

### What happened?

The presence of default `SvgTheme` argument of svg output prevents `DefaultSvgTheme` in `flutter_svg` works correctly.

```dart
SVG.cube.svg(width: 64, theme: SvgTheme(currentColor: Colors.red)),
SvgPicture.asset('assets/svg/cube.svg',
    width: 64, theme: SvgTheme(currentColor: Colors.red)),
```

image

```dart
child: DefaultSvgTheme(
  theme: SvgTheme(currentColor: Colors.red),
  child: Column(
    children: [
      SVG.cube.svg(width: 64),
      SvgPicture.asset('assets/svg/cube.svg', width: 64),
```

image

### Relevant a pubspec.yaml.

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mym0404**
The default SvgTheme argument prevents `DefaultSvgTheme` provider works correctly. This should be null and works as passthrough behavior

## What does this change?

Remove default `SvgTheme` from svg generator

Fixes #484 🎯

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [x] Make sure to open a GitHub issue as a bug/feature request before writing your code! That way we can discuss the change, evaluate designs, and agree on the general idea
  - [x] Ensure the tests (`melos run test`)
  - [x] Ensure the analyzer and formatter pass (`melos run format` to automatically apply formatting)
- [ ] Appropriate docs were updated (if necessary)

## Result

I can confirm that it works after patch this.

```yaml
  flutter_gen_runner:
    git:
      url: https://github.com/mym0404/flutter_gen.git
      path: packages/runner
      ref: core-patched
```

### CODE REVIEW

**Comment by AlexV525** [comment]
Thanks for the contribution, could you also add tests indicate that `DefaultSvgTheme` works?

**Comment by mym0404** [comment]
@AlexV525 You mean should we add `flutter` and `flutter_test` dev dependencies into `core` package?

File: examples/example/pubspec.yaml [hunk 1]
```diff
@@ -22,6 +22,9 @@ dev_dependencies:
   lints: ^2.0.0
   build_runner: ^2.0.0
   flutter_gen_runner: ^5.4.0
+  test: ^1.16.0
```
Comment (AlexV525) [inline examples/example/pubspec.yaml:25]: Should we remove this?

**Review by AlexV525 — APPROVED** [review]
The fix looks good so far!

File: examples/example/pubspec.yaml (hunk 1 above)
Comment (mym0404) [reply examples/example/pubspec.yaml:25]: Sure I will

Comment (mym0404) [reply examples/example/pubspec.yaml]: Done

**Pull request merged**